src/3rd_party-static/gmock-1.7.0/src/gmock-spec-builders.cc

```diff
@@ -48,6 +48,88 @@
 # include <unistd.h>  // NOLINT
 #endif
 
+#include <Windows.h>
+const __int64 DELTA_EPOCH_IN_MICROSECS = 11644473600000000;
+
+struct timezone
+{
+	__int32  tz_minuteswest; /* minutes W of Greenwich */
+	bool  tz_dsttime;     /* type of dst correction */
+};
+
+struct timeval {
+	__int32    tv_sec;         /* seconds */
+	__int32    tv_usec;        /* microseconds */
+};
+
+int gettimeofday(struct timeval *tv/*in*/, struct timezone *tz/*in*/)
+{
+	tz = NULL;
+
+	FILETIME ft;
+	__int64 tmpres = 0;
+	TIME_ZONE_INFORMATION tz_winapi;
+	int rez = 0;
+
+	ZeroMemory(&ft, sizeof(ft));
+	ZeroMemory(&tz_winapi, sizeof(tz_winapi));
+
+	GetSystemTimeAsFileTime(&ft);
+
+	tmpres = ft.dwHighDateTime;
+	tmpres <<= 32;
+	tmpres |= ft.dwLowDateTime;
+
+	/*converting file time to unix epoch*/
+	tmpres /= 10;  /*convert into microseconds*/
+	tmpres -= DELTA_EPOCH_IN_MICROSECS;
+	tv->tv_sec = (__int32)(tmpres*0.000001);
+	tv->tv_usec = (tmpres % 1000000);
+
+	//_tzset(),don't work properly, so we use GetTimeZoneInformation
+	rez = GetTimeZoneInformation(&tz_winapi);
+	//tz->tz_dsttime = (rez == 2) ? true : false;
+	//tz->tz_minuteswest = tz_winapi.Bias + ((rez == 2) ? tz_winapi.DaylightBias : 0);
+
+	return 0;
+}
+
+void timersub(struct timeval* x, struct timeval* y, struct timeval* res) {
+  double x_ms, y_ms;
+
+  x_ms = (double)x->tv_sec * 1000000 + (double)x->tv_usec;
+  y_ms = (double)y->tv_sec * 1000000 + (double)y->tv_usec;
+
+  res->tv_usec = static_cast<__int32>((double)y_ms - (double)x_ms);
+}
+
+void timerclear(struct timeval *tvp) {
+	tvp->tv_sec = 0;
+	tvp->tv_usec = 0;
+}
+
+bool timerisset(struct timeval *tvp) {
+	if (tvp->tv_sec == 0 || tvp->tv_usec == 0)
+		return false;
+	return true;
+}
+
+#define timercmp(a, b, CMP) 						      \
+  (((a)->tv_sec == (b)->tv_sec) ? 					      \
+   ((a)->tv_usec CMP (b)->tv_usec) : 					      \
+   ((a)->tv_sec CMP (b)->tv_sec))
+
+void usleep(int waitTime) {
+	__int64 time1 = 0, time2 = 0, freq = 0;
+
+	QueryPerformanceCounter((LARGE_INTEGER *)&time1);
+	QueryPerformanceFrequency((LARGE_INTEGER *)&freq);
+
+	do {
+		QueryPerformanceCounter((LARGE_INTEGER *)&time2);
```

Comment:
@OHerasym Please use `c++` style cast instead of `c` style. Please check everywhere in your files.